app/models/page.rb

```diff
@@ -2,6 +2,9 @@ class Page < ApplicationRecord
   has_paper_trail
 
   belongs_to :form
+  has_many :routing_conditions, class_name: "Condition", foreign_key: "routing_page_id", dependent: :destroy
+  has_many :check_conditions, class_name: "Condition", foreign_key: "check_page_id", dependent: :nullify
+  has_many :goto_conditions, class_name: "Condition", foreign_key: "goto_page_id", dependent: :nullify
```

Comment:
Here there are a set of associations for page objects to conditions - you can find all the conditions taking place on a given page with `p.routing_conditions`, all the conditions that check this page with `p.check_conditions` and all the conditions that direct towards this page with `p.goto_conditions` (I'm definitely open to naming suggestions to make this less confusing)

I did initially set it up so that you could do `p.conditions` to return associated conditions, but it's really messy when you're getting back conditions that can be referring to this page in different capacities. It's something we could add if it turns out it's valuable though.

Reply:
Don't think we need to change anything in this PR, but just noting that we'll have to do some eager loading to make these associations available in the API/Admin